airflow-core/src/airflow/api_fastapi/core_api/routes/public/dags.py

```diff
@@ -54,23 +54,45 @@
 from airflow.api_fastapi.core_api.datamodels.dags import (
     DAGCollectionResponse,
     DAGDetailsResponse,
+    DAGFavoriteBody,
     DAGPatchBody,
     DAGResponse,
 )
 from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
 from airflow.api_fastapi.core_api.security import (
     EditableDagsFilterDep,
+    GetUserDep,
     ReadableDagsFilterDep,
     requires_access_dag,
 )
 from airflow.api_fastapi.logging.decorators import action_logging
 from airflow.exceptions import AirflowException, DagNotFound
 from airflow.models import DAG, DagModel
+from airflow.models.dag_favorite import DagFavorite
 from airflow.models.dagrun import DagRun
 
 dags_router = AirflowRouter(tags=["DAG"], prefix="/dags")
 
 
+@dags_router.get("/favorite", dependencies=[Depends(requires_access_dag(method="GET"))])
+def get_favorite_dags(session: SessionDep, user: GetUserDep) -> DAGCollectionResponse:
+    """Get DAGs favorited by the user."""
```

Comment:
```suggestion
    """Get DAGs favorited by the user."""
```
Reword, `favorited` seems weird